HEC.FDA.StatisticsTests/Histograms/HistogramTests.cs

```diff
@@ -11,518 +11,832 @@
+        Assert.True(errorStd < tolerance);
+    }
+
+    [Fact]
+    public void HistogramResetToZeroPreservesDistribution()
+    {
+        Triangular triangular = new Triangular(min: 0.0001, mode: 100, max: 1000);
+        DynamicHistogram histogram = new DynamicHistogram(min: triangular.Min, binWidth: 2, new ConvergenceCriteria());
+        Random random = new Random(Seed: 1234);
+        int sampleSize = 50000;
+        double[] sample = new double[sampleSize];
+        for (int i = 0; i < sampleSize; i++)
         {
-            IDistribution stdNormal = new Statistics.Distributions.Normal(mean, standardDeviation);
-            var rand = new Random();
-            Histogram histogram = new Histogram(null, binWidth);
-            double[] data = new double[n];
-            for (int i = 0; i < n; i++)
-            {
-                var randProb = rand.NextDouble();
-                data[i] = stdNormal.InverseCDF(randProb);
-            }
-            histogram.AddObservationsToHistogram(data);
-            double actualMean = histogram.Mean;
-            double meanErr = Math.Abs((expectedMean - actualMean) / actualMean);
+            sample[i] = (triangular.InverseCDF(random.NextDouble()));
+        }
+        histogram.AddObservationsToHistogram(sample);
+        double zeroVal = 0;
+        histogram.AddObservationToHistogram(zeroVal);
 
-            double actualMedian = histogram.InverseCDF(.5);
-            double medianErr = Math.Abs((expectedMedian - actualMedian) / actualMedian);
 
-            double errTol = 0.01;
-            Assert.True(meanErr < errTol);
-            Assert.True(medianErr < errTol);
+        Assert.Equal(zeroVal, histogram.Min);
 
+        double tolerance = 0.06;
+        int steps = 100;
+        for (int i = 0; i < steps; i++)
+        {
+            double probabilityStep = i + 1;
+            double probability = (probabilityStep) / steps;
+            double histogramQuantile = histogram.InverseCDF(probability);
+            double triangularQuantile = triangular.InverseCDF(probability);
+            double relativeDifference = Math.Abs((histogramQuantile - triangularQuantile) / triangularQuantile);
+            Assert.True(relativeDifference < tolerance);
         }
-        */
+    }
 
-        [Fact]
-        public void AddingHistograms()
-        {
-            Normal normal1 = new Normal(3, 1);
-            Normal normal2 = new Normal(4, 2);
-            Normal normal3 = new Normal(2, 5);
-            double[] probabilities = new double[] { .25, .5, .75 };
-            double[] expected = new double[probabilities.Length];
-            for (int i = 0; i < probabilities.Length; i++)
-            {
-                expected[i] = normal1.InverseCDF(probabilities[i]);
-                expected[i] += normal2.InverseCDF(probabilities[i]);
-                expected[i] += normal3.InverseCDF(probabilities[i]);
-            }
+    #region GitHub Ticket #1236 - EqAD with Lateral Structures   
```

Comment:
reference to ticket?

Reply:
I think it's relatively helpful, until we're long confident this issue is in the past.